README.md

```diff
@@ -125,7 +142,19 @@ Current state | Next word | Next state
 
 ## A2.4
 
-«replace this with your answer»
+Current state | Next word | Next state
+--------------|-----------|-----------
+    S0        |    the    |     S1
+	S1        |           |     S2
+	S1        |    lazy   |     S1
+	S1        |   smelly  |     S1
+	S2        |    dog    |     S3
+	S2        |    cat    |     S3
+	S3        |    ate    |     S4
+	S3        |    ran    |     S4
+	S4        |           |     EOI
+	S4        |   slowly  |     EOI
+	S4        |  noisily  |     EOI
```

Comment:
# 4

Two problems here. First, there's no empty transition in a state machine, at least not the way its written here. In the grammar it means "match nothing", but there's no corresponding event.

Also, EOI is an event, not a state, so lines 155-157 aren't right.